search/matcher.go

```diff
@@ -0,0 +1,80 @@
+package search
+
+import (
+	"strings"
+
+	"github.com/launchdarkly/ld-find-code-refs/flags"
+	"github.com/launchdarkly/ld-find-code-refs/internal/helpers"
+	"github.com/launchdarkly/ld-find-code-refs/internal/ld"
+	"github.com/launchdarkly/ld-find-code-refs/internal/log"
+	"github.com/launchdarkly/ld-find-code-refs/options"
+)
+
+type ElementMatcher struct {
+	Elements   []string
+	Aliases    map[string][]string
+	Delimiters []string
+	ProjKey    string
+	Directory  string
+}
+
+type Matcher struct {
+	Elements   []ElementMatcher
+	CtxLines   int
+	Delimiters string
+}
+
+// Scan checks the configured directory for flags base on the options configured for Code References.
+func Scan(opts options.Options, repoParams ld.RepoParams) (Matcher, []ld.ReferenceHunksRep) {
+	flagMatcher := ElementMatcher{
+		Directory: opts.Dir,
+	}
+	flagMatcher.Elements, flagMatcher.Aliases = flags.GenerateSearchElements(opts, repoParams)
+
+	matcher := Matcher{
+		Elements: []ElementMatcher{flagMatcher},
+		CtxLines: opts.ContextLines,
+	}
+
+	// Configure delimiters
+	delims := getDelimiters(opts)
+	matcher.Delimiters = strings.Join(helpers.Dedupe(delims), "")
```

Comment:
Does it make sense to detect duplicate delimiters early and fail early as part of config validation? I see we are already validating the configured directory in `NormalizeAndValidatePath`, we can expand this to include duplicate delimiters.

Reply:
I'll note this down for a follow up PR. Given the size of this one currently if I can get it merged in these changes should be much easier.